### DISCUSSION TITLE
Date Component Entagngle Issue

### DISCUSSION THREAD

**Discussion by robs007** [livewire/livewire#4443] [1. Help]
I'm using the screencast for data tables from the Livewire site. I want to use the date component in my project. I can not get that component to work with any of my dates from my model. Whether I use it directly or with an accessor. I get this alpine error.

**cdn.min.js:1 Alpine Expression Error: Cannot read properties of null (reading 'date_for_editing')**

However, if I create a new public property and set it to carbon date, either Cardon::now() or Carbon::now()->format('m/d/Y') it works fine and I don't get the error. Of course, that breaks my model binding, so when I go to do a save it doesn't work. I have tried all the date fields in my model and I get the same results. If I change the component to a text component it displays the date correctly, so I know the model and accessor are working correctly. I'm new to livewire and alpine. 
Thanks
 Here is the code in the component.
`<div
    x-data="{ value: @entangle($attributes->wire('model')), picker: undefined }"
    x-init="new Pikaday({ field: $refs.input, format: 'MM/DD/YYYY', onOpen() { this.setDate($refs.input.value) } })"
    x-on:change="value = $event.target.value"
    class="flex rounded-md shadow-sm"
>
    <span class="inline-flex items-center px-3 rounded-l-md border border-r-0 border-gray-300 bg-gray-50 text-gray-500 sm:text-sm">
        <svg class="h-5 w-5 text-gray-400" fill="currentColor" viewBox="0 0 20 20">
            <path fill-rule="evenodd" clip-rule="evenodd" d="M6 2C5.44772 2 5 2.44772 5 3V4H4C2.89543 4 2 4.89543 2 6V16C2 17.1046 2.89543 18 4 18H16C17.1046 18 18 17.1046 18 16V6C18 4.89543 17.1046 4 16 4H15V3C15 2.44772 14.5523 2 14 2C13.4477 2 13 2.44772 13 3V4H7V3C7 2.44772 6.55228 2 6 2ZM6 7C5.44772 7 5 7.44772 5 8C5 8.55228 5.44772 9 6 9H14C14.5523 9 15 8.55228 15 8C15 7.44772 14.5523 7 14 7H6Z"/>
        </svg>
    </span>

    <input
        {{ $attributes->whereDoesntStartWith('wire:model') }}
        x-ref="input"
        x-bind:value="value"
        class="form-control"
    />
</div>`

`            <x-input.group for="date_for_editing" label="Date" :error="$errors->first('editing.date_for_editing')">
                <x-input.date wire:model="date_for_editing" id="date_for_editing" />
            </x-input.group>
`

**Comment by jonneroelofs** [comment]
https://laravel-livewire.com/docs/2.x/properties#binding-models

Did you add your model bound dates to the rules? See  relevant docs in link above.

**Comment by reneecst** [comment]
Try wire:model="editing.date_for_editing"
